packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -244,6 +245,25 @@ export class PureDataObjectFactory<
 		return this.createNonRootInstanceCore(runtime, [this.type], initialState);
 	}
 
+	public async createInstance2(
```

Comment:
I like everything but the name :)
This is just an API, and not on compat boundary of any sort. Can we simply reuse existing createInstance and change its return type? All clients would need to update (which is a minus), but at least we have good long-term naming.
Otherwise, I assume we will eventually remove createInstance() and will leave with createInstance2()

That all said, Win32 uses this pattern (most calls functions Ex suffix, though there is one API that has "2" at the end :)).
That said, it usually does so when "extended" function takes way more arguments than "simple" version, usually return types are the same.